src/components/Table/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { composeTableHead } from '../TableHead';
+import './styles.css';
+
+export enum SortDirection {
+  asc = 'ASC',
+  desc = 'DESC',
+}
+
+export type ColumnType<T extends string | number> = {
+	dataName: T
+	displayName: string
+	hidden?: boolean
+}
+
+export interface Props<T extends string | number> {
+	columns: ColumnType<T>[]
+	defaultSortColumn: T
+	searchTerm: string
+	searchColumn: T
+	headClasses?: string
+	onSearch: (arg: string) => void
+	renderTableContent: (...arg: any[]) => any
+}
+
+export const composeTable = <T extends string | number>() => {
```

Comment:
I find the `composeTable` wrapper a bit clunky, but I suppose there's not much we can do about that to still enforce the generic Enum type for all `T` fields. I'd probably add this to the bottom of the file, so people don't have to use the `composeTable` for non-enum (and exclusively JS) tables 
```js
export const Table = composeTable<string>();
```

Reply:
If only you could generically type JSX!